### PR TITLE
#161744188 fix social auth

### PR DESCRIPTION
**Pull request by caveinn**
#### What does this PR do?
- Fixes a bug on social auth

#### Description of Task to be completed?
- Fixed an error that caused a user to not get authenticated since it required

#### How should this be manually tested?
- clone this repo  `git clone https://github.com/andela/ah-shakas.git` 
- cd into it `cd ah-shakas`
- checkout to this branch `git checkout bg-fix-social-auth-161744188`
- install dependencies and activate virtual env `pipenv shell` then `pipenv install`
- run the server `./manage.py runserver`
- in your postman put the following data depending the provider you are using:
                      {
    "provider": "facebook",
    "access_token": "access token"
    
}
#### Any background context you want to provide?
- when using twitter authentication,apart from "access_token" you also need "access_token_secret" because it uses oauth1

#### What are the relevant pivotal tracker stories?

[#161744188](https://www.pivotaltracker.com/n/projects/2198533/stories/161744188)

### CODE REVIEW

**Comment by lukorito** [comment]
Hi @caveinn its not advisable to include access tokens in the PR, maybe you can save it as an env variable

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -243,21 +243,28 @@ def create(self, request, *args, **kwargs):
             return Response({"error": "The email is already registered, please try another one"},
                             status=status.HTTP_400_BAD_REQUEST)
 
-        if user and user.is_active:
+        if user:
```
Comment (Arthur236) [inline authors/apps/authentication/views.py:246]: Why did you remove the user.is_active check?
Shouldnt this be restricted to users that have actually activated their accounts?

Comment (caveinn) [reply authors/apps/authentication/views.py:246]: The is active part was for checking that the user has confirmed their email. But since we are using social authentication there is no need to confirm your email since it has to be valid to be working on any of the platforms.

**Comment by caveinn** [comment]
@Kitika the access token was for a  test user. It is not an actual token.

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (Arthur236) [reply authors/apps/authentication/views.py:246]: Alright. That makes sense

**Review by Arthur236 — APPROVED** [review]
Looks good 👍

**Review by michael-basweti — APPROVED** [review]
I have tested it,looks good
